azure-kusto-data/azure/kusto/data/__init__.py

```diff
@@ -1,4 +1,148 @@
-from .kusto_client import KustoClient, KustoResultIter, KustoResponse
-from .kusto_exceptions import KustoError, KustoClientError, KustoServiceError
+from enum import Enum
+from six import text_type
 from .version import VERSION
 __version__ = VERSION
+
+class KustoConnectionStringBuilder(dict):
+    class _Keywords(Enum):
+        data_source = "Data Source"
+        aad_user_id = "AAD User ID"
+        password = "Password"
+        application_client_id = "Application Client Id"
+        application_key = "Application Key"
+        authority_id = "Authority Id"
+
+    def __init__(self, connection_string, *args, **kwargs):
+        self._constructing = True
+        if connection_string is not None and '=' not in connection_string.split(';')[0]:
+            connection_string = 'Data Source=' + connection_string
+        for kvp_string in connection_string.split(';'):
+            kvp = kvp_string.split('=')
+            self[kvp[0]] = kvp[1]
+        super(KustoConnectionStringBuilder, self).__init__(*args, **kwargs)
+        self._is_valid()
+        self._constructing = False
+
+    def __setitem__(self, key, value):
+        try:
+            keyword = KustoConnectionStringBuilder._Keywords(key)
+        except ValueError:
+            raise ValueError("%s is not supported as an item in KustoConnectionStringBuilder" % key)
```

Comment:
This suggests to me that this shouldn't be a dict and you should just have a normal class or provide functions that operate directly on a dict as necessary.